Assets/Scripts/Gameplay/AI/Systems/Attacking/ObserveArea.cs

```diff
@@ -26,9 +35,53 @@ private void OnTriggerExit(Collider other)
         {
             if (other.gameObject.GetComponent<Player>() && _targetsTransforms.Count != 0)
             {
-                _targetsTransforms.Clear();
+                _targetsTransforms.Remove(other.gameObject.transform);
                 TargetsChanged?.Invoke();
             }
         }
+
+        private void Start()
+        {
+            _lastPosition = transform.position;
+        }
+        
+        private void RotationRight()
+        {
+            _rotationTween = transform.DOLocalRotate(new Vector3(0f, _angle, 0f), _rotationSpeed, RotateMode.Fast)
+                .OnComplete(RotationLeft);
+        }
+        
+        private void RotationLeft()
+        {
+            _rotationTween = transform.DOLocalRotate(new Vector3(0f, -_angle, 0f), _rotationSpeed, RotateMode.Fast)
+                .OnComplete(RotationRight);
+        }
+        
+        private void KillRotationTween()
+        {
+            _rotationTween.Kill();
+            transform.rotation = new Quaternion();
+        }
+
+        private void Update()
+        {
+            _isMoving = Mathf.Abs(transform.position.magnitude - _lastPosition.magnitude) > MovementThreshold ;
```

Comment:
Pls change _isMoving var from private field to local var. ANd send this var to ProcessRotation() method

Reply:
Done